### PR TITLE
CONTRIBUTING.md ahead of move of repo to XRPLF

### PR DESCRIPTION
**Pull request by RichardAH**
## High Level Overview of Change
Add a `CONTRIBUTING.md` ahead of moving the repository to the XRP Ledger Foundation.

If you are listed as a maintainer and do not wish to be listed please comment.

Suggestions / changes welcome.

### CODE REVIEW

**Review by nbougalis — APPROVED** [review]
This looks good and I personally don't want to bikeshed too much on this; even if there improvements, they could be done in a separate PR, but a `CONTRIBUTING.md` is needed and has been missing for a long time.

Two brief comments (food for thought, really): consider whether some of the docs under `docs/` should be linked here.

I know that @wojake took a swipe at this with [PR #3955](https://github.com/ripple/rippled/pull/3955), and that spurred me to also [tackle this](https://gist.github.com/nbougalis/fc27cd877ff39d098352f81a050a8ab1). Not suggesting either of those are better (mine certainly isn't; it's overly full of emojis! :vomiting_face:), but I decided to link those previous efforts here for inspiration.

:+1: from me, as is.

**Comment by nbougalis** [comment]
I'm following up here but not because I think we need to refine this further; as I said during my review, this looks good to me, especially as a first step.

But I ~suspect~hope that this will be a living document, and so with that in mine, I wanted to link Bitcoin's [`CONTRIBUTING.md`](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md), which is more fleshed out (but also not _that_ different) here as a guidepost for future improvements.

**Comment by RichardAH** [comment]
It's definitely intended to be improved / a living document. I think the main focus at the moment is getting the repo into the foundation ASAP, and further policy refinements can be made after

**Comment by wojake** [comment]
>I wanted to link Bitcoin's [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md)

Please consider adding a `"decision making" process` section in this file's initial commit. 

I've had a hard time finding the right people to peer review my material in the past, I don't know how hard and lengthy of a process it is to peer review code which configures a decentralized payment network, so I leave this decision to Richard and the others.

[Bitcoin Core's CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#decision-making-process) copy looks good to me as a starting point, _hopefully_ we'll tweak it as we progress.

**Comment by nbougalis** [comment]
> > I wanted to link Bitcoin's [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md)
> 
> Please consider adding a `"decision making" process` section in this file's initial commit.
> 
> I've had a hard time finding the right people to peer review my material in the past, I don't know how hard and lengthy of a process it is to peer review code which configures a decentralized payment network, so I leave this decision to Richard and the others.

It's hard to know who the right person to review code is. Again, looking at Bitcoin, they have a [file](https://github.com/bitcoin/bitcoin/blob/master/REVIEWERS) that lists reviewers with expertise in the various parts of the code. It would be good to think about adopting this too.

Generally, it's hard to know how long a review takes. It's a process, and it's not always easy or clean and it's rarely pleasant. Even @JoelKatz once complained that he's having a hard time getting his code to pass code review. And that is _not_ necessarily a bad thing, as long the review process is objective.

> [Bitcoin Core's CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#decision-making-process) copy looks good to me as a starting point, _hopefully_ we'll tweak it as we progress.

While that part of the contribution guidelines addresses something different, in reading through your comment I did have a thought that I wanted to raise.

@RichardAH, I wholeheartedly agree with your comment that "the main focus" should be on "getting the repo into the foundation ASAP, and further policy refinements can be made after." However, I think that there's a small change we should consider adding _now_ rather than later:

I'd change the section about maintainers to include a process for adding new and removing existing maintainers:


> ## Maintainers
> 
> Maintainers are ecosystem participants with elevated access to the repository and are able to push new code, make decision on whether a release should be made, etc.
> 
> ### Process
> 
> New maintainers can be proposed by two existing maintainers, subject to a vote by a quorum of the existing maintainers. A minimum of 50% support _and_ a 50% participation is required. In the event of a tie vote, the addition of the new maintainer will be rejected.
> 
> Existing maintainers can resign or, be subject to a vote for removal at the behest of two existing maintainers. A minimum of 60% support _and_ a 50% participation is required. The XRP Ledger Foundation will have the ability, for cause, to initiate the removal of an existing maintainer and such a request is not subject to a vote.
> 
> ### Existing Maintainers
> * [JoelKatz](https://github.com/JoelKatz) (Ripple)
> * [Manojsdoshi](https://github.com/manojsdoshi) (Ripple)
> * [N3tc4t](https://github.com/n3tc4t) (XRPL Labs)
> * [Nikolaos D Bougalis](https://github.com/nbougalis) (Ripple)
> * [Nixer89](https://github.com/nixer89) (XRP Ledger Foundation)
> * [RichardAH](https://github.com/RichardAH) (XRPL Labs + XRP Ledger Foundation)
> * [Seelabs](https://github.com/seelabs) (Ripple)
> * [Silkjaer](https://github.com/Silkjaer) (XRP Ledger Foundation)
> * [WietseWind](https://github.com/WietseWind) (XRPL Labs + XRP Ledger Foundation)
> * [Ximinez](https://github.com/ximinez) (Ripple)
>

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# Maintainers
+Maintainers are ecosystem participants with elevated access to the repository and are able to push new code, make decision on whether a release should be made, etc.
+
+## Code Review
+New contributors' PRs must be reviewed by at least two of the maintainers. Well established prior contributors can be reviewed by a single maintainer.
+
+## Adding and Removing
+New maintainers can be proposed by two existing maintainers, subject to a vote by a quorum of the existing maintainers. A minimum of 50% support and a 50% participation is required. In the event of a tie vote, the addition of the new maintainer will be rejected.
+
+Existing maintainers can resign or, be subject to a vote for removal at the behest of two existing maintainers. A minimum of 60% support and a 50% participation is required. The XRP Ledger Foundation will have the ability, for cause, to initiate the removal of an existing maintainer and such a request is not subject to a vote.
```
Comment (intelliot) [inline CONTRIBUTING.md:55]:
```suggestion
Existing maintainers can resign, or be subject to a vote for removal at the behest of two existing maintainers. A minimum of 60% agreement and 50% participation are required. The XRP Ledger Foundation will have the ability, for cause, to remove an existing maintainer without a vote.
```

**Review by intelliot — APPROVED** [review]
Just some minor suggestions; looks good overall

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# Contributing
+The XRP Ledger has many and diverse stakeholders, and everyone deserves a chance to contribute meaningful changes to the code that runs the XRPL.
+To contribute, please:
+1. Fork the repository under your own user.
+2. Create a new branch on which to write your changes. Please note that changes which alter the state or function of the ledger itself must be composed via and guarded using [Amendments](https://xrpl.org/amendments.html). Changes which are _read only_ i.e. RPC, or changes which are only refactors and maintain the existing behaviour do not need to be made through an Amendment.
```
[the remark this answers is not in the document]
Comment (RichardAH) [reply CONTRIBUTING.md:5]: An amendment can change, for example, consensus itself (as in the `NegativeUNL` amendment.) I can change it if you want but it is reducing a more correct statement to a less correct statement.

[the remark this answers is not in the document]
Comment (MarkusTeufelberger) [reply CONTRIBUTING.md:5]: `NegativeUNL` introduced new transaction types, if it were implemented without on-ledger objects, it wouldn't have needed an amendment. I agree about changing state though, so a middle ground could be "changes which alter transaction processing and/or ledger state"?

**Pull request merged**